com.unity.shadergraph/Editor/Data/Graphs/ColorShaderProperty.cs

```diff
@@ -9,9 +9,26 @@ namespace UnityEditor.ShaderGraph.Internal
     [FormerName("UnityEditor.ShaderGraph.ColorShaderProperty")]
     public sealed class ColorShaderProperty : AbstractShaderProperty<Color>
     {
+        public override int latestVersion => 1;
+
+        [SerializeField]
+        private int m_Version = 0;
+
+        public override int version
+        {
+            get => m_Version;
+            protected set => m_Version = value;
+        }
+
         internal ColorShaderProperty()
         {
             displayName = "Color";
+            version = 0;
```

Comment:
agreed

Reply:
ah okay now I remember why I didnt do that - so this is a new value that has not yet been serialized. That means opening up new graphs, their default value is going to be assumed to be `1` since it will call the constructor from deserialization, then populate what it knows about (in our case, not version). 
Although, in this case, it may be preferable to say all deserialized properties are created at lowest version, and if a value is serialized it will be replaced

Reply:
but, to do this, I would either need to create a special case for the color property in deserialization, or figure out a generic way to handle this for any property/jsonObject

Reply:
New thought, move version number/base version change calls to JsonObject. Makes it easier to deprecate/change version of anything, and can explicitly set version to 0 in deserialization unless otherwise specified. 
thoughts?

Reply:
Yes, last thing sounds good. Admitedly I haven't thought enough about it but I like it in principle. =)